src/input_common/gcadapter/gc_poller.cpp

```diff
@@ -0,0 +1,301 @@
+// Copyright 2020 yuzu Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#include <atomic>
+#include <list>
+#include <mutex>
+#include <utility>
+#include "common/threadsafe_queue.h"
+#include "input_common/gcadapter/gc_adapter.h"
+#include "input_common/gcadapter/gc_poller.h"
+
+namespace InputCommon {
+
+class GCButton final : public Input::ButtonDevice {
+public:
+    explicit GCButton(int port_, int button_, int axis_,
+                      std::shared_ptr<GCAdapter::Adapter> adapter)
```

Comment:
```suggestion
                      GCAdapter::Adapter* adapter)
```

Given this class uses the pointer as a non-owning pointer